src/ifcopenshell-python/ifcopenshell/validate.py

```diff
@@ -365,7 +412,61 @@ def validate(f, logger, express_rules=False):
 
         if not has_invalid_value:
             for i, (attr, val, is_derived) in enumerate(zip(attrs, values, entity.derived())):
-
+                try:
+                    type_of_attribute = attr.type_of_attribute()
+                    declared_type = type_of_attribute.declared_type()
+                    type_name = declared_type.name()
+
+                    if type_name == "IfcDateTime" and val:
+
+                        parsing = parse_datetime(val)
+
+                        if isinstance(parsing, DateErrorParsing().__class__):
```

Comment:
```suggestion
                        if isinstance(parsing, DateErrorParsing):
```

DateErrorParsing().__class__ == DateErrorParsing